### PR TITLE
BVPS-292 GC Notify Regenerate PIN

### PR DESCRIPTION
**Pull request by emilymendelson**

### CODE REVIEW

File: src/tests/routes/pins.spec.ts [hunk 1]
```diff
@@ -810,9 +916,9 @@ describe('Pin endpoints', () => {
         const reqBody = validCreatePinBodySinglePidServiceBC;
         const res = await request(app).post('/pins/regenerate').send(reqBody);
         expect(res.statusCode).toBe(500);
-        expect(res.body.message).toBe(
-            `Cannot read properties of undefined (reading 'metadata')`,
-        );
+        // expect(res.body.message).toBe(
```
Comment (hannah-macdonald1) [inline src/tests/routes/pins.spec.ts:1012]: This can be removed if it's no longer needed

File: src/tests/routes/pins.spec.ts [hunk 2]
```diff
@@ -419,9 +459,9 @@ describe('Pin endpoints', () => {
         const reqBody = validCreatePinBodySinglePid;
         const res = await request(app).post('/pins/vhers-create').send(reqBody);
         expect(res.statusCode).toBe(500);
-        expect(res.body.message).toBe(
-            `Cannot read properties of undefined (reading 'metadata')`,
-        );
+        // expect(res.body.message).toBe(
```
Comment (hannah-macdonald1) [inline src/tests/routes/pins.spec.ts:527]: This can be removed

File: src/controllers/pinController.ts [hunk 3]
```diff
@@ -711,6 +752,7 @@ export class PINController extends Controller {
      */
     private async createOrRecreatePinServiceBC(
         requestBody: serviceBCCreateRequestBody,
+        regenerateOrCreate: string,
```
Comment (hannah-macdonald1) [inline src/controllers/pinController.ts:758]: Only problem with this is that you'll have to update the FE to actually use the correct endpoint, and we have to assume that whatever has been sent is correct... you may get a regenerate email when you were actually creating and vice versa

**Review by hannah-macdonald1 — COMMENTED** [review]
Left a few comments. Let me know what you think

File: src/controllers/pinController.ts (hunk 3 above)
Comment (emilymendelson) [reply src/controllers/pinController.ts:758]: oooooh good point, let me fix that

File: src/controllers/pinController.ts [hunk 4]
```diff
@@ -749,11 +793,16 @@ export class PINController extends Controller {
...
-        const errors = await batchUpdatePin(
+
+        const batchUpdatePinResponse = await batchUpdatePin(
             [pinResult[0]],
             emailPhone,
             requestBody.requesterUsername, // TODO: Get info from token
         );
+
+        const errors = batchUpdatePinResponse[0];
+        const regenerateOrCreate = batchUpdatePinResponse[1];
```
Comment (emilymendelson) [inline src/controllers/pinController.ts:804]: @hannah-macdonald1 Updated so that regenerate/create is coming from batchUpdatePinResponse function

**Review by hannah-macdonald1 — APPROVED** [review]
Looks good!

**Pull request merged**